### PR TITLE
Cmake changes

### PR DESCRIPTION
**Pull request by tobylane**
Vcpkg seems to require the with_sdl setting.

SDLMain is different in SDL2, the folder isn't used.

### CODE REVIEW

**Comment by tobylane** [comment]
> This is suspect. Never had an issue on Windows without it. What were you seeing?

> CMake Error at CorsixTH/CMakeLists.txt:83 (find_package):
  Could not find a package configuration file provided by "SDL2" with any of
  the following names:
    SDL2Config.cmake
    sdl2-config.cmake
  Add the installation prefix of "SDL2" to CMAKE_PREFIX_PATH or set
  "SDL2_DIR" to a directory containing one of the above files.  If "SDL2"
  provides a separate development package or SDK, be sure it has been
  installed.
-- Configuring incomplete, errors occurred!

**Comment by TheCycoONE** [comment]
VcPkg knows nothing about "WITH_SDL": https://github.com/microsoft/vcpkg/search?q=WITH_SDL&unscoped_q=WITH_SDL that's a parameter that we made up for CorsixTH.

I'm guessing a coincidence with a dependency not installed the first time and available the second time. Try again without WITH_SDL.

**Comment by tobylane** [comment]
Fixed, it was the vcpkg cache.

File: CorsixTH/CMakeLists.txt [hunk 1]
```diff
@@ -110,13 +104,21 @@ find_package(Lua REQUIRED)
 if(Lua_FOUND)
   target_link_libraries(CorsixTH ${LUA_LIBRARY})
   include_directories(${LUA_INCLUDE_DIR})
-  # Special link flags needed on OSX/64bit, according to: http://luajit.org/install.html
+  # Special link flags needed on macOS, according to: http://luajit.org/install.html
   # If these are not specified, luaL_newstate() returns NULL and we get this:
   #   Fatal error starting CorsixTH: Cannot open Lua state.
-  if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin" AND LUA_INTERPRETER_TYPE STREQUAL "LuaJIT" AND CMAKE_SIZEOF_VOID_P EQUAL 8)
+  if(APPLE AND LUA_LIBRARIES MATCHES "libluajit")
```
Comment (TheCycoONE) [inline CorsixTH/CMakeLists.txt]: Should have been called out that you're dropping compiling for 32bit macOS (I know Catalina drops it, but not everyone is on Catalina).

Alternately leave in the `CMAKE_SIZEOF_VOID_P` condition.

Comment (tobylane) [reply CorsixTH/CMakeLists.txt]: 10.6, Snow Leopard, was the last to support 32bit hardware. I can leave it in, for people who intentionally built 32bit lua?

Comment (TheCycoONE) [reply CorsixTH/CMakeLists.txt]: Might as well.

**Pull request merged**